src/content/developers/docs/data-structures/rlp/index.md

```diff
@@ -0,0 +1,161 @@
+
+3.  the data is a string if the range of the first byte is [0xb8, 0xbf], and the length of the string whose length in bytes is equal to the first byte minus 0xb7 follows the first byte, and the string follows the length of the string;
+
+4.  the data is a list if the range of the first byte is [0xc0, 0xf7], and the concatenation of the RLP encodings of all items of the list which the total payload is equal to the first byte minus 0xc0 follows the first byte;
+
+5.  the data is a list if the range of the first byte is [0xf8, 0xff], and the total payload of the list whose length is equal to the first byte minus 0xf7 follows the first byte, and the concatenation of the RLP encodings of all items of the list follows the total payload of the list;
+
+In code, this is:
+
+```python
+def rlp_decode(input):
+    if len(input) == 0:
+        return
+    output = ''
+    (offset, dataLen, type) = decode_length(input)
+    if type is str:
+        output = instantiate_str(substr(input, offset, dataLen))
+    elif type is list:
+        output = instantiate_list(substr(input, offset, dataLen))
+    output + rlp_decode(substr(input, offset + dataLen))
+    return output
+
+def decode_length(input):
+    length = len(input)
+    if length == 0:
+        raise Exception("input is null")
+    prefix = ord(input[0])
+    if prefix <= 0x7f:
+        return (0, 1, str)
+    elif prefix <= 0xb7 and length > prefix - 0x80:
+        strLen = prefix - 0x80
+        return (1, strLen, str)
+    elif prefix <= 0xbf and length > prefix - 0xb7 and length > prefix - 0xb7 + to_integer(substr(input, 1, prefix - 0xb7)):
+        lenOfStrLen = prefix - 0xb7
+        strLen = to_integer(substr(input, 1, lenOfStrLen))
+        return (1 + lenOfStrLen, strLen, str)
+    elif prefix <= 0xf7 and length > prefix - 0xc0:
+        listLen = prefix - 0xc0;
+        return (1, listLen, list)
+    elif prefix <= 0xff and length > prefix - 0xf7 and length > prefix - 0xf7 + to_integer(substr(input, 1, prefix - 0xf7)):
+        lenOfListLen = prefix - 0xf7
+        listLen = to_integer(substr(input, 1, lenOfListLen))
+        return (1 + lenOfListLen, listLen, list)
+    else:
+        raise Exception("input don't conform RLP encoding form")
+
+def to_integer(b):
+    length = len(b)
+    if length == 0:
+        raise Exception("input is null")
+    elif length == 1:
+        return ord(b[0])
+    else:
+        return ord(substr(b, -1)) + to_integer(substr(b, 0, -1)) * 256
+```
+
+## Further reading {#further-reading}
+
+- [RLP in Ethereum](https://medium.com/coinmonks/data-structure-in-ethereum-episode-1-recursive-length-prefix-rlp-encoding-decoding-d1016832f919)
+- [Ethereum under the hood: RLP](https://medium.com/coinmonks/ethereum-under-the-hood-part-3-rlp-decoding-df236dc13e58)
+
+## Related topics {#related-topics}
+
+- [Patricia merkle trie](/developers/docs/data-structures/patricia-merkle-trie)
```

Comment:
We're mixing usage of "trie" and "tries"... 

The other link to this page uses `patricia-merkle-tries`, but currently the path is `patricia-merkle-trie`.

Would suggest we just go with the pluralized form, which means updating the new markdown file path to `/patricia-merkle-tries/index.md` and then update this link:
```suggestion
- [Patricia merkle trie](/developers/docs/data-structures/patricia-merkle-tries)
```